kernel/serial_port/src/lib.rs

```diff
@@ -186,12 +186,20 @@ impl SerialPort {
                 match SerialPortAddress::try_from(base_port) {
                     Ok(SerialPortAddress::COM1 | SerialPortAddress::COM3) => {
                         INTERRUPT_ACTION_COM1_COM3.call_once(|| 
-                            Box::new(move || { deferred_task.unblock(); })
+                            Box::new(move || {
+                                deferred_task
+                                    .unblock()
+                                    .expect("BUG: com_1_com3_interrupt_handler: couldn't unblock deferred task");
+                            })
```

Comment:
i find this kind of formatting really tedious and hard to read. Personally would prefer something like:
```rust
Box::new(move || deferred_task.unblock()
    .expect(...)
)
```

Reply:
ah i see. It's not because it's multiple lines, it's because we don't wan't to return the value of the last expression in the block, we want the return to be void `()`.

anyway, good enough.